src/domain/usecases/GetNotificationsUseCase.ts

```diff
@@ -0,0 +1,17 @@
+import _ from "lodash";
+
+import { NotificationRepository } from "../repositories/NotificationRepository";
+import { FutureData } from "../types/Future";
+import { Notification } from "../entities/Notification";
+
+export class GetNotificationsUseCase {
+    constructor(private notificationRepository: NotificationRepository) {}
+
+    public execute(): FutureData<Notification[]> {
+        return this.notificationRepository.list(null).map(notifications =>
```

Comment:
I know `notificationRepository.list` is not strictly related to this PR, but it's a recent method, let's align with our typical approach, we may use an object with optional props instead of Maybe.